### PR TITLE
Feature/M2-2024 Builder - Flanker - Practice and Test Round Settings

### PR DESCRIPTION
**Pull request by IhorNesterenko99**
image

### CODE REVIEW

File: src/modules/Builder/features/PerformanceTasks/Flanker/RoundSettings/BlockSequences/BlockSequencesContent/BlockSequencesContent.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import { Svg, Table, UiType } from 'shared/components';
+import {
+  StyledBodyMedium,
+  StyledSvgPrimaryColorBtn,
+  StyledTitleMedium,
+  theme,
+  variables,
+} from 'shared/styles';
+import { useCurrentActivity } from 'modules/Builder/hooks';
+import { FlankerStimulusSettings } from 'modules/Builder/pages/BuilderApplet/BuilderApplet.types';
```
Comment (kdanilenka32) [inline src/modules/Builder/features/PerformanceTasks/Flanker/RoundSettings/BlockSequences/BlockSequencesContent/BlockSequencesContent.tsx:15]: I suggest to move the type to modules/Builder/types/

File: src/modules/Builder/features/PerformanceTasks/Flanker/RoundSettings/BlockSequences/BlockSequencesContent/BlockSequencesContent.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    if (!uploadedTable?.length) return;
+    const blocks = getRoundBlocks(stimulusTrials, uploadedTable);
+    if (!blocks) return;
+
+    setValue(blockSequencesField, blocks);
+  }, [uploadedTable, stimulusTrials]);
+
+  return stimulusTrials?.some((trial) => !!trial.imageName) ? (
+    <>
+      <Box>
```
Comment (kdanilenka32) [inline src/modules/Builder/features/PerformanceTasks/Flanker/RoundSettings/BlockSequences/BlockSequencesContent/BlockSequencesContent.tsx:59]: Do y need the Box?

**Review by muhurov — APPROVED** [review]

File: src/modules/Builder/features/PerformanceTasks/Flanker/RoundSettings/BlockSequences/BlockSequencesContent/BlockSequencesContent.utils.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import i18n from 'i18next';
+import { HeadCell } from 'shared/types';
+import { createArray, getTableCell } from 'shared/utils';
+import { FlankerStimulusSettings } from 'modules/Builder/pages/BuilderApplet/BuilderApplet.types';
+import { Row } from 'shared/components';
+import { UploadedTable } from './BlockSequencesContent.types';
```
Comment (kdanilenka32) [inline src/modules/Builder/features/PerformanceTasks/Flanker/RoundSettings/BlockSequences/BlockSequencesContent/BlockSequencesContent.utils.ts:6]: space

File: src/modules/Builder/features/PerformanceTasks/Flanker/RoundSettings/BlockSequences/BlockSequencesContent/ImportSequencesPopup/ImportSequencesPopup.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    <StyledBodyLarge>{t('flankerRound.successfullyUploaded')}</StyledBodyLarge>,
+    <StyledBodyLarge color={variables.palette.semantic.error}>
+      {t(`flankerRound.${isUpload ? 'uploadError' : 'updateError'}`)}
+    </StyledBodyLarge>,
+  ];
+
+  const screens = getScreens(isUpload, components);
+
+  const onSubmit = async () => {
+    if (step === 0) {
```
Comment (kdanilenka32) [inline src/modules/Builder/features/PerformanceTasks/Flanker/RoundSettings/BlockSequences/BlockSequencesContent/ImportSequencesPopup/ImportSequencesPopup.tsx:55]: better to use switch

File: src/modules/Builder/features/PerformanceTasks/Flanker/RoundSettings/BlockSequences/BlockSequencesContent/ImportSequencesPopup/ImportSequencesPopup.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      onClose={onClose}
+      onSubmit={onSubmit}
+      title={t(
+        `${isUpload ? 'flankerRound.uploadBlockSequences' : 'flankerRound.updateBlockSequences'}`,
+      )}
+      buttonText={t(screens[step].btnText)}
+      hasSecondBtn={screens[step].hasSecondBtn}
+      secondBtnText={t(screens[step].secondBtnText || '')}
+      onSecondBtnSubmit={onClose}
+      disabledSubmit={isSubmitDisabled}
```
[the remark this answers is not in the document]
Comment (IhorNesterenko99) [reply src/modules/Builder/features/PerformanceTasks/Flanker/RoundSettings/BlockSequences/BlockSequencesContent/ImportSequencesPopup/ImportSequencesPopup.tsx:93]: if file was uploaded, then on step 0 submit is not disabled

File: src/modules/Builder/features/PerformanceTasks/Flanker/RoundSettings/BlockSequences/BlockSequencesContent/BlockSequencesContent.tsx (hunk 2 above)
Comment (IhorNesterenko99) [reply src/modules/Builder/features/PerformanceTasks/Flanker/RoundSettings/BlockSequences/BlockSequencesContent/BlockSequencesContent.tsx:59]: yes for btn to not occupy the full width of the section

**Review by kdanilenka32 — APPROVED** [review]

**Pull request merged**